ntrans_gui.py

```diff
@@ -200,29 +236,42 @@ def generate_ntrans_dictionary(self) -> None:
             "target_language": self.target_language_var.get(),
         }
 
-        for key in user_choices:
-            if not user_choices[key]:
-                print(
-                    "All fields need to be filled in before generating the N-Trans dictionary."
-                )  # TODO: Change print to label
-                break  # TODO: break needs to change to `return`. Will be fixed in #23
+        for value in self.user_choices.values():
+            if not value:
+                self.setting_not_defined_warning.grid()
+                return
+        self.setting_not_defined_warning.grid_remove()
+
+    def get_save_file_path(self) -> None:
+        savepath = tkinter.filedialog.askdirectory()  # type: ignore
+        if savepath:
+            self.filepath.set(savepath)
+
+    def generate_ntrans_dictionary(self) -> None:
+
+        # TEST
+        for key, value in self.user_choices.items():
+            if not value:
+                self.flash_warning_label()
+                return
 
         # TODO: Check filepath for validity
         if self.thread is None:
-            self.progress_queue: queue.Queue[int] = queue.Queue()
+            self.progress_queue = queue.Queue()
             # Calls logic in ntrans.py
             self.thread = threading.Thread(
-                target=ntrans.read_ngram_files, args=[user_choices, self.progress_queue]
+                target=ntrans.read_ngram_files,
+                args=[self.user_choices, self.progress_queue],
             )
             self.thread.start()
 
-            self.estimated_time_label.grid_remove()
-            self.progress_frame.grid(
-                column=0, row=16, columnspan=2, padx=(0, 0), pady=(20, 40)
-            )
-            self.check_progressbar_queue()
+        self.estimated_time_label.grid_remove()
+        self.progress_frame.grid(
+            column=0, row=16, columnspan=2, padx=(0, 0), pady=(20, 40)
+        )
+        self.check_progressbar_queue()
```

Comment:
Is there a reason why indentation changed?

Reply:
merge issue